src/zenml/cli/base.py

```diff
@@ -64,18 +68,48 @@ def init(path: Optional[Path]) -> None:
 
 
 @cli.command("clean")
-@click.option("--yes", "-y", type=click.BOOL, default=False)
+@click.option("--yes", "-y", is_flag=True, default=False)
 def clean(yes: bool = False) -> None:
-    """Clean everything in repository.
+    """Delete all ZenML metadata and artifacts.
+
+    This is a destructive operation, primarily intended for use in development.
 
     Args:
       yes: bool:  (Default value = False)
     """
     if not yes:
-        _ = confirmation(
-            "This will completely delete all pipelines, their associated "
-            "artifacts and metadata ever created in this ZenML repository. "
+        confirm = confirmation(
+            "DANGER: This will completely delete all artifacts and metadata ever created \n"
+            "in this ZenML repository. Note: Pipelines and stack components running \n"
+            "non-locally will still exist. Please delete them manually.\n"
             "Are you sure you want to proceed?"
         )
 
-    error("Not implemented for this version")
+    if yes or confirm:
+        local_zen_repo_config = Path.cwd() / REPOSITORY_DIRECTORY_NAME
+        global_zen_config = Path(get_global_config_directory())
+        if fileio.exists(str(local_zen_repo_config)):
+            fileio.rmtree(str(local_zen_repo_config))
+            declare(f"Deleted local ZenML config from {local_zen_repo_config}.")
+        if fileio.exists(str(global_zen_config)):
+            config_yaml_path = global_zen_config / CONFIG_FILE_NAME
+            config_yaml_data = yaml_utils.read_yaml(str(config_yaml_path))
+            old_user_id = config_yaml_data["user_id"]
+            for dir_name in fileio.listdir(str(global_zen_config)):
+                if fileio.isdir(str(global_zen_config / str(dir_name))):
+                    warning(
+                        f"Deleting '{str(dir_name)}' directory from global config."
+                    )
+            fileio.rmtree(str(global_zen_config))
+            declare(f"Deleted global ZenML config from {global_zen_config}.")
+
+        Repository.initialize(root=Path.cwd())
+        if old_user_id:
+            new_config_yaml_data = yaml_utils.read_yaml(str(config_yaml_path))
+            new_config_yaml_data["user_id"] = old_user_id
+            yaml_utils.write_yaml(str(config_yaml_path), new_config_yaml_data)
```

Comment:
I don't think I understand what this does: you're basically reading and re-writing the same file contents you had before the CLI command was run.

I think you have to remove the `profiles` field, otherwise you end up with the same profiles as before, but it would be even better to do this via the GlobalConfiguration class itself instead of manipulating raw yaml files here. Calling the GlobalConfiguration constructor with custom values will create a new object that, in the absence of a configuration yaml file, will act as if you're running ZenML for the first time (which is what you want). The only issue is that you have to preserve the `user_id` and `analytics_opt_in` values somehow, which might require some small alterations to the GlobalConfiguration.__init__ method to allow you to supply them as constructor args.